### PR TITLE
[chore] Allow pinging code owners on PRs

### PR DESCRIPTION
**Pull request by evan-bradley**
**Description:**

Adds code owners without write permissions as reviewers to a PR when the corresponding component label is added to the PR.

**Link to tracking Issue:**

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16036

**Testing:**

I tested this to a certain degree in my personal fork. I wasn't able to test some of the specifics around adding code owners as reviewers (e.g. adding someone who has already been added), so this will have to be tested on actual Collector contrib PRs.

### CODE REVIEW

File: .github/workflows/ping-codeowners-issues.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: 'Ping code owners on issues'
```
Comment (TylerHelmuth) [inline .github/workflows/ping-codeowners-issues.yml:1]: Is this essentially a rename that calls the script that was renamed?

**Review by TylerHelmuth — COMMENTED** [review]
I still support this method (manually adding labels) as an initial step towards full automation.  Here is a previous PR with background on the topic: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14347

File: .github/workflows/ping-codeowners-issues.yml (hunk 1 above)
Comment (evan-bradley) [reply .github/workflows/ping-codeowners-issues.yml:1]: Yes. The one change is that I did [remove the label filter](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16175/files?short_path=fb0fd06#diff-fb0fd06da4d79a15e0d66efd4c1c24e84ad41d37b152f4bcb7ea1e218d547072L8) to run the workflow. I'm open to re-adding it if we want it, but I figure it's relatively a minor inefficiency for not having to maintain the list of component types.

**Review by mx-psi — APPROVED** [review]

File: .github/workflows/ping-codeowners-issues.yml (hunk 1 above)
Comment (TylerHelmuth) [reply .github/workflows/ping-codeowners-issues.yml:1]: Ya I'm good with that

**Review by TylerHelmuth — APPROVED** [review]
Will adding component labels via comments work on PRs?  Would be nice if users could select a label during PR creation like they can for issues.

**Comment by evan-bradley** [comment]
> Will adding component labels via comments work on PRs?

This isn't supported right now, GitHub makes it nontrivial (though not super difficult) due to the way workflows and the CLI work. I'm thinking we will probably be okay without it, but if there's demand I think it's certainly possible.

> Would be nice if users could select a label during PR creation like they can for issues.

This would be a little challenging since issue forms aren't supported in PRs. We could still parse the title/body for components, but I'm working on something that just looks at the changed files to automatically determine this.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**